compiler/src/iree/compiler/Codegen/Common/TransformDialectJitterPass.cpp

```diff
@@ -391,31 +448,114 @@ class TransformDialectJitterPass
     linalg::registerTransformDialectExtension(registry);
   }
 
-  TransformDialectJitterPass() = default;
+  LogicalResult buildStrategy(ImplicitLocOpBuilder ib, Value variantH,
+                              StringRef target, StringRef strategyName,
+                              ArrayRef<int64_t> workgroupSize) {
+    // TODO: this is very brittle.
+    if (target == "cuda") {
+      if (strategyName == "reduction_strategy")
+        return buildReductionCudaStrategy(ib, variantH, workgroupSize);
+    } else if (target == "llvm-cpu") {
+      if (strategyName == "reduction_strategy")
+        return buildReductionCpuStrategy(ib, variantH, workgroupSize);
+    }
+    return getOperation()->emitError("Unknown strategy: ") << strategyName;
+  }
+
+  // TODO: this is very brittle.
+  FailureOr<StringRef> getTarget(HAL::ExecutableVariantOp variantOp) {
```

Comment:
Instead of implementing all these here reuse from https://github.com/iree-org/iree/blob/main/compiler/src/iree/compiler/Codegen/Utils/Utils.h

Also the TODO not very meaningful....